src/zkevm_specs/evm/execution/error_Invalid_jump.py

```diff
@@ -0,0 +1,53 @@
+from ...util import FQ
+from ..instruction import Instruction, Transition
+from ..table import CallContextFieldTag
+from ..execution_state import ExecutionState
+from ..opcode import Opcode
+from ...util import N_BYTES_PROGRAM_COUNTER
+
+
+def invalid_jump(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+    # current executing op code must be JUMP or JUMPI
+    instruction.constrain_in(opcode, [FQ(Opcode.JUMP), FQ(Opcode.JUMPI)])
+    code_length = instruction.bytecode_length(instruction.curr.code_hash)
+    dest = instruction.stack_pop()
+    # lookup value from bytecode table
+    dest_value = instruction.rlc_to_fq(dest, N_BYTES_PROGRAM_COUNTER)
+
+    out_of_range, _ = instruction.compare(code_length, dest_value, N_BYTES_PROGRAM_COUNTER)
+
+    # if not out of range, check `dest` is invalid
+    if not out_of_range.n:
+        value, is_code = instruction.bytecode_lookup_pair(instruction.curr.code_hash, dest_value)
+        # value is not `JUMPDEST` or `is_code` is false
+        instruction.constrain_bool(is_code)
+        is_jump_dest = value == Opcode.JUMPDEST
+        instruction.constrain_zero(is_code * FQ(is_jump_dest))
+    else:
+        # `dest` value is out of range
+        instruction.constrain_equal(out_of_range, FQ(1))
+
+    # current call must be failed.
+    is_success = instruction.call_context_lookup(CallContextFieldTag.IsSuccess)
+    instruction.constrain_equal(is_success, FQ(0))
+
+    # Go to EndTx only when is_root
+    is_to_end_tx = instruction.is_equal(instruction.next.execution_state, ExecutionState.EndTx)
+    instruction.constrain_equal(FQ(instruction.curr.is_root), is_to_end_tx)
+
+    if instruction.curr.is_root:
+        # Do step state transition
+        instruction.constrain_step_state_transition(
+            rw_counter=Transition.delta(2 + instruction.curr.reversible_write_counter),
```

Comment:
```suggestion
            rw_counter=Transition.delta(1 + instruction.curr.reversible_write_counter),
```

Reply:
we have two rw lookups : 1 stack lookup + 1 callcontext lookup, so here is 2 delta of rw counter. any reason for only 1 required ?

Reply:
Sorry I forgot about stack read, `2` is correct!